pkg/spdk/engine.go

```diff
@@ -634,6 +658,67 @@ func (e *Engine) handleUblkFrontend(spdkClient *spdkclient.Client) (err error) {
 	return nil
 }
 
+func (e *Engine) reconnectNvmeTcpFrontend(spdkClient *spdkclient.Client) (err error) {
+	if e.NvmeTcpFrontend == nil {
+		return fmt.Errorf("cannot reconnect: NvmeTcpFrontend is nil")
+	}
+
+	if e.NvmeTcpFrontend.IP == "" || e.NvmeTcpFrontend.Port == 0 {
+		return fmt.Errorf("NvmeTcpFrontend IP or Port is not set")
+	}
+
+	if e.initiator == nil {
+		return fmt.Errorf("initiator is missing for reconnecting frontend")
+	}
+
+	defer func() {
+		if err == nil {
+			if errUpdateLogger := e.log.UpdateLogger(logrus.Fields{
+				"endpoint":   e.Endpoint,
+				"port":       e.NvmeTcpFrontend.Port,
+				"targetPort": e.NvmeTcpFrontend.TargetPort,
+			}); errUpdateLogger != nil {
+				e.log.WithError(err).Warn("Failed to update logger during reconnect frontend")
+			}
+			e.log.Infof("Finished reconnecting frontend for engine: %+v", e)
+		}
+	}()
+
+	// Stop previous target (if it exists)
+	// This ensures that any previous exported bdev target with the same NQN is cleanly removed
+	// from the SPDK target subsystem. SPDK does not allow re-exporting a bdev with the same NQN
+	// without first removing the old instance. Skipping this step may result in:
+	// Therefore, this is a defensive cleanup step to make the target namespace idempotent.
```

Comment:
This explains why we need to stop the old nvmf target. But it is not related to why the nvmf target is stopped before the nvmf initiator disconnection. 
Or in other words, can we do something like: `e.initiator.DisconnectNVMeTCPTarget` -> `spdkClient.StopExposeBdev` -> `spdkClient.StartExposeBdev` -> `e.initiator.StartNvmeTCPInitiator`